state/metrics.go

```diff
@@ -10,6 +10,12 @@ import (
 var meter = otel.Meter("state")
 
 func WithMetrics(ca *CoreAccessor) {
+	if ca == nil {
+		// short-circuit if node is running without state access
+		log.Error("node is running without state access, cannot enable metrics for state module")
+		return
+	}
```

Comment:
There is a way to make FX not call this func when CoreAccess does not exist. Also this logs will be annoying, imo